### PR TITLE
Update the name of the path in HDF sres parameter

### PR DESCRIPTION
**Pull request by jwlodek**
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
During testing of ophyd-async devices, I was seeing a traceback caused by a KeyError on the `path` key. This is because in newer versions of ophyd-async the stream resource uses `dataset` as the parameter key name, rather than `path`: https://github.com/bluesky/ophyd-async/blob/5ac94e283b170b603df42a7d5fe33fd7f2786524/src/ophyd_async/core/_hdf_dataset.py#L64

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

```
doc = {'uid': '17a66a91-2b4a-40ae-b9be-97921a94bcd1', 'data_key': 'manta-cam1', 'mimetype': 'application/x-hdf5', 'uri': 'file://localhost/ns
ls2/data/tst/legacy/mock-proposals/2024-1/pass-000000/assets/proj_1b2430d3-2c75-48c2-8267-7bab4e20d63f.h5', 'parameters': {'dataset': '/entry
/data/data', 'swmr': False, 'multiplier': 1}, 'run_start': '98494492-75ff-49f2-a332-41d8e3fcd065'}                                           
```

Without this change, an `RE(count([manta]))` will fail with a KeyError:

```
    246 self._handlers[sres_uid] = handler                                                                                                   
    247 self._sres_nodes[sres_uid] = sres_node                                                                                               
                                                                                                                                             
File ~/.ipython/profile_collection_tst/overlays/bluesky/src/bluesky/consolidators.py:186, in ConsolidatorBase.get_data_source(self=<bluesky.c
onsolidators.HDF5Consolidator object>)                                                                                                       
    170 def get_data_source(self) -> DataSource:                                                                                             
    171     """Return a DataSource object reflecting the current state of the streamed dataset.                                              
    172                                                                                                                                      
    173     The returned DataSource is conceptually similar (and can be an instance of) tiled.structures.DataSource. In                      
    174     general, it describes associated Assets (filepaths, mimetype) along with their internal data structure                           
    175     (array shape, chunks, additional parameters) and should contain all information necessary to read the file.                      
    176     """                                                                                                                              
    177     return DataSource(                                                                                                               
    178         mimetype=self.mimetype,                                                                                                      
    179         assets=self.assets,                                                                                                          
    180         structure_family=StructureFamily.array,                                                                                      
    181         structure=ArrayStructure(                                                                                                    
    182             data_type=BuiltinDtype.from_numpy_dtype(self.dtype),                                                                     
    183             shape=self.shape,                                                                                                        
    184             chunks=self.chunks,                                                                                                      
    185         ),                                                                                                                           
--> 186         parameters=self.adapter_parameters,                                                                                          
        self.mimetype = 'application/x-hdf5'                                                                                                 
        self = <bluesky.consolidators.HDF5Consolidator object at 0x7f6ac7cd5a10>                                                             
        self.assets = [Asset(data_uri='file://localhost/nsls2/data/tst/legacy/mock-proposals/2024-1/pass-000000/assets/proj_92a13da7-bcb2-448
f-ac77-630d456d5c6e.h5', is_directory=False, parameter='data_uri', num=None, id=None)]                                                       
        StructureFamily.array = <StructureFamily.array: 'array'>                                                                             
        self.dtype = dtype('float64')                                                                                                        
        Management.external = <Management.external: 'external'>                                                                              
    187         management=Management.external,                                                                                              
    188     )                                                                                                                                
                                                                                                                                             
File ~/.ipython/profile_collection_tst/overlays/bluesky/src/bluesky/consolidators.py:242, in HDF5Consolidator.adapter_parameters(self=<bluesk
y.consolidators.HDF5Consolidator object>)                                                                                                    
    235 @property                                                                                                                            
    236 def adapter_parameters(self) -> Dict:                                                                                                
    237     """Parameters to be passed to the HDF5 adapter, a dictionary with the keys:                                                      
    238                                                                                                                                      
    239     path: List[str] - file path represented as list split at `/`                                                                     
    240     swmr: bool -- True to enable the single writer / multiple readers regime                                                         
    241     """                                                                                                                              
--> 242     return {"path": self._sres_parameters["path"].strip("/").split("/"), "swmr": self.swmr}                                          
        self.swmr = False                                                                                                                    
        self = <bluesky.consolidators.HDF5Consolidator object at 0x7f6ac7cd5a10>                                                             
        self._sres_parameters = {'dataset': '/entry/data/data', 'swmr': False, 'multiplier': 1}                                              
                                                                                                                                             
KeyError: 'path'                                                                                                                             

```

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Running unit tests, & running against a camera in our test lab.

<!--
## Screenshots (if appropriate):
-->

### CODE REVIEW

**Comment by jwlodek** [comment]
One note is that this parameter is almost certainly used somewhere else downstream (`tiled`?), and that should probably be adjusted as well.

**Comment by jwlodek** [comment]
Tests are expected to fail because after this change is applied `TiledWriter` tests will fail until: https://github.com/bluesky/tiled/pull/775 is merged.

File: src/bluesky/consolidators.py [hunk 1]
```diff
@@ -236,10 +236,10 @@ def __init__(self, stream_resource: StreamResource, descriptor: EventDescriptor)
     def adapter_parameters(self) -> Dict:
         """Parameters to be passed to the HDF5 adapter, a dictionary with the keys:
 
-        path: List[str] - file path represented as list split at `/`
+        dataset: List[str] - file dataset represented as list split at `/`
```
Comment (genematx) [inline src/bluesky/consolidators.py:239]:
```suggestion
        dataset: List[str] - a path to the dataset within the hdf5 file represented as list split at `/`
```

**Review by genematx — APPROVED** [review]
Thanks for updating it! Just one minor suggestion from me.

**Comment by danielballan** [comment]
Power-cycled to re-run CI with latest tiled release.

**Review by danielballan — APPROVED** [review]

**Comment by jwlodek** [comment]
Merging since all the tests are passing

**Pull request merged**